modules/compress/src/main/java/org/apache/ignite/internal/processors/compress/CompressionProcessorImpl.java

```diff
@@ -170,6 +177,28 @@ private ByteBuffer doCompactPage(ByteBuffer page, int pageSize) throws IgniteChe
         return compactPage;
     }
 
+    /** Check if filesystem actually supports punching holes. */
+    private void checkPunchHole(Path storagePath, int fsBlockSz) throws IgniteException {
+        Path testFile = storagePath.resolve("punch_hole_" + UUID.randomUUID() + ".test");
+
+        ByteBuffer buffer = GridUnsafe.allocateBuffer(fsBlockSz * 2);
+        GridUnsafe.zeroMemory(GridUnsafe.bufferAddress(buffer), buffer.capacity());
+
+        try {
+            try (RandomAccessFileIO testFileIO = new RandomAccessFileIO(testFile.toFile(), CREATE, WRITE)) {
+                testFileIO.writeFully(buffer);
+
+                testFileIO.punchHole(fsBlockSz, fsBlockSz);
+            }
+            finally {
+                Files.deleteIfExists(testFile);
+            }
+        }
+        catch (Exception e) {
```

Comment:
I think `LinkageError` should be caught too.

Reply:
It cannot be thrown here, it may be thrown at static initializer

Reply:
```
/** */
    static {
        NativeFileSystem x = null;

        try {
            if (IgniteComponentType.COMPRESSION.inClassPath()) {
                if (U.isLinux())
                    x = U.newInstance(NATIVE_FS_LINUX_CLASS);
            }
        }
        catch (Throwable e) {
            err = e;
        }

        fs = x;
    }
```
in `FileSystemUtils`